consensus/src/twins_test.rs

```diff
@@ -200,3 +235,193 @@ fn basic_start_test() {
         );
     });
 }
+
+/// This test checks that the split_network function works
+/// as expected, that is: nodes in a partition with less nodes
+/// than required for quorum do not commit anything.
+///
+/// Setup:
+///
+/// 4 honest nodes (n0, n1, n2, n3), and 0 twins.
+/// Create two partitions p1=[n1], and p2=[n0, n2, n3] with
+/// a proposer in p2.
+///
+/// Test:
+///
+/// Run consensus for enough rounds to potentially form a commit.
+/// Check that n1 has no commits, and n0 has commits.
+///
+/// Run the test:
+/// cargo xtest -p consensus drop_config_test -- --nocapture
+#[test]
+fn drop_config_test() {
+    let mut runtime = consensus_runtime();
+    let mut playground = NetworkPlayground::new(runtime.handle().clone());
+    let num_nodes = 4;
+    let num_twins = 0;
+    let mut nodes =
+        SMRNode::start_num_nodes_with_twins(num_nodes, num_twins, &mut playground, FixedProposer);
+
+    // 4 honest nodes
+    let n0 = nodes[0]
+        .config
+        .validator_network
+        .as_ref()
+        .unwrap()
+        .peer_id();
+    let n1 = nodes[1]
+        .config
+        .validator_network
+        .as_ref()
+        .unwrap()
+        .peer_id();
+    // n2 will be the leader (because it has the lowest hash)
```

Comment:
If we change the hashing algorithm, this might not always be true.  Might be best to calculate this on the fly.

Reply:
True. This comment was from previous code which needed to identify the proposer. It is no longer relevant -- removed.